lib/ibm_watson/watson_service.rb

```diff
@@ -59,11 +60,14 @@ def initialize(vars)
       end
     end
 
-    if !vars[:iam_access_token].nil? || !vars[:iam_apikey].nil?
+    if @icp_prefix
+      @username = "apikey"
+      @password = vars[:iam_apikey]
+    elsif !vars[:iam_access_token].nil? || !vars[:iam_apikey].nil?
       _token_manager(iam_apikey: vars[:iam_apikey], iam_access_token: vars[:iam_access_token], iam_url: vars[:iam_url])
     elsif !vars[:username].nil? && !vars[:password].nil?
       if vars[:username] == "apikey"
```

Comment:
I think the check for the ICP prefix should be here, since users are going to be passing in
```
username: "apikey"
password: "icp-<rest of key>"
```

With that check, it should catch that case and go to basic auth.

Also, make sure to look for the prefix "icp-", including the dash.

Reply:
What about the case when users pass in 
```
apikey: "icp-key"
```
Do we need to check for that or no? Because general instructions allow the user to pass the `iam_apikey` when instantiating, and that case will fail. Although, I am not sure ICP will give credentials to users as `username`/`password` or just give them an `apikey`.